### PR TITLE
date-range-filter

### PR DESCRIPTION
**Pull request by RashmiDholakiya**
Implement global date range filter.

### CODE REVIEW

File: app/Http/Controllers/FilterController.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function dateFilter(Request $req)
+    {
+
+        session(['from_date' => $req['f'] , 'to_date' => $req['t']]);
+
+        echo session('from_date');
```
Comment (poojajajal) [inline app/Http/Controllers/FilterController.php]: What is use of this ?

Comment (RashmiDholakiya) [inline app/Http/Controllers/FilterController.php]: For debug purpose I have echo the dates.

File: app/Http/Controllers/FilterController.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        echo session('to_date');
+    }
+    public function index()
```
Comment (poojajajal) [inline app/Http/Controllers/FilterController.php]: Please remove all method which have a no use.

File: app/Http/Controllers/ProjectController.php [hunk 3]
```diff
@@ -17,120 +18,120 @@
...
-    }
+	/**
+	 * Display a listing of the resource.
+	 * @return Response
+	 */
+	public function index()
+	{
+
+		if (session('from_date') != NULL) {
+			$projects = Project::whereBetween('start_date', [session('from_date'), session('to_date')])->paginate(15);
```
Comment (poojajajal) [inline app/Http/Controllers/ProjectController.php]: Paginate number add into constant.

File: app/Http/Controllers/FilterController.php (hunk 2 above)
Comment (RashmiDholakiya) [inline app/Http/Controllers/FilterController.php]: Done

File: app/Http/Controllers/ProjectController.php [hunk 4]
```diff
@@ -17,120 +18,120 @@
...
+	 * Display the specified resource.
+	 * @param  int $id
+	 * @param RevenueVsCost $revenue_vs_cost
+	 * @return Response
+	 */
+	public function show($id, RevenueVsCost $revenue_vs_cost)
+	{
+		$project     = Project::with(['revenues', 'costs', 'costs.staff'])
+			->findOrFail($id);
+		$performance = $revenue_vs_cost->get(['project_id' => $id], ['month_logged']);
```
Comment (poojajajal) [inline app/Http/Controllers/ProjectController.php]: Why you need to use two separate array ?

File: public/js/custom/filter.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * Created by rashmi-dholakiya on 23/3/16.
+ */
+$(document).ready(function () {
+
+    $("#datepickerFrom").datepicker({
+        dateFormat: 'yy-mm-dd',
+        onSelect: function (dateText, inst) {
+            if ($('#datepickerTo').val() != '') {
+                if ($('#datepickerTo').val() < $('#datepickerFrom').val()) {
+                    alert('Your To-Date must be greater then From-Date');
```
Comment (poojajajal) [inline public/js/custom/filter.js]: Don't use alert, there is so many plug-in available to display any alert message so please use it.

File: public/js/custom/filter.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+    });
+
+    function store_date(from, to) {
+        var base_url = window.location.origin;
+        $.ajax({
+            type: 'POST',
+            url: base_url + "/folio/public/filter",
+            data: {f: from, t: to},
```
Comment (poojajajal) [inline public/js/custom/filter.js]: What do you mean by f and t ? Please use proper naming convention.

File: public/js/custom/filter.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+                }
+                else {
+                    store_date($("#datepickerFrom").val(), $("#datepickerTo").val());
+                }
+            }
+        }
+    });
+
+    function store_date(from, to) {
+        var base_url = window.location.origin;
```
Comment (poojajajal) [inline public/js/custom/filter.js]: You need project base url right ? This is not good way .
On master file you can do something like this 
 <script type="text/javascript">
        var base_url = {!! "'".URL::to('/')."/'" !!};
    </script>

File: public/js/custom/filter.js (hunk 6 above)
Comment (RashmiDholakiya) [inline public/js/custom/filter.js]: Done

File: app/Http/Controllers/ProjectController.php (hunk 3 above)
Comment (RashmiDholakiya) [inline app/Http/Controllers/ProjectController.php]: Done

File: public/js/custom/filter.js (hunk 5 above)
Comment (RashmiDholakiya) [inline public/js/custom/filter.js]: I have added the 'SweetAlert' plugin for alerts.

File: public/js/custom/filter.js (hunk 7 above)
Comment (RashmiDholakiya) [inline public/js/custom/filter.js]: Done

File: public/js/custom/filter.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                }
+            }
+        }
+    });
+
+    $("#datepickerTo").datepicker({
+        dateFormat: 'yy-mm-dd',
+        onSelect: function (dateText, inst) {
+            if ($('#datepickerFrom').val() == '') {
+                alert('Select From date');
```
Comment (poojajajal) [inline public/js/custom/filter.js]: Change alert.

Comment (RashmiDholakiya) [inline public/js/custom/filter.js]: Done.

File: public/js/custom/filter.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Created by rashmi-dholakiya on 23/3/16.
+ */
+$(document).ready(function () {
+
+    $("#datepickerFrom").datepicker({
+        dateFormat: 'yy-mm-dd',
+        onSelect: function (dateText, inst) {
+            if ($('#datepickerTo').val() != '') {
```
Comment (poojajajal) [inline public/js/custom/filter.js]: Can we use just one if condition here  ?

Comment (RashmiDholakiya) [inline public/js/custom/filter.js]: Yeah Done

**Comment by poojajajal** [comment]
Code look good now.

File: app/Http/Controllers/FilterController.php [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+
+class FilterController extends Controller
+{
+    public function dateFilter(Request $req)
```
Comment (poojajajal) [inline app/Http/Controllers/FilterController.php]: Change variable name to $request.

File: app/Http/Controllers/ProjectController.php [hunk 11]
```diff
@@ -17,120 +18,120 @@
...
-    }
+	/**
+	 * Display a listing of the resource.
+	 * @return Response
+	 */
+	public function index()
+	{
+
+		if (session('from_date') != NULL) {
+			$projects = Project::whereBetween('start_date', [session('from_date'), session('to_date')])->paginate(PAGINATE_LIMIT);
```
Comment (poojajajal) [inline app/Http/Controllers/ProjectController.php]: Get this data from  ProjectRepository  and no need of this if else condition you can do like this below.
ProjectContoller 
$from_date = (session('from_date') != NULL)?session('from_date'):' ';
$projects = $this->project_repository->get($from_date)
ProjectRepository 
public function get($from_date = ' '){
Project:: where(function($q){
if($from_date != ""){
$q->whereBetween('start_date', [session('from_date'), session('to_date')]);
}
)->paginate(PAGINATE_LIMIT);

}

Comment (RashmiDholakiya) [inline app/Http/Controllers/ProjectController.php]: It is already done in latest code.

File: app/Http/Controllers/FilterController.php (hunk 10 above)
Comment (RashmiDholakiya) [inline app/Http/Controllers/FilterController.php]: It is also changed in latest code

**Pull request merged**